src/biokbase/narrative/tests/test_appmanager.py

```diff
@@ -1107,118 +1113,100 @@ def test_transform_input_bad(self):
         with self.assertRaises(ValueError):
             app_util.transform_param_value("foo", "bar", None)
 
-    def _verify_comm_mock(
-        self,
-        send_comm_mock: MagicMock,
-        num_calls: int,
-        expected_messages: List[str] = None,
-        expected_keys: List[List[str]] = None,
-        expected_values: List[List[dict]] = None,
-    ) -> None:
-        """
-        Validates the usage of the MagicMock used instead of sending a comm message to the
-        front end. This is expected to mock any calls to JobComm.mock_comm_channel. Each of
-        the "expected_*" kwargs are given as a list - each element represents which time the
-        call was made, in order.
-
-        This is done through a series of asserts. If they all pass and None is returned, then
-        the mock was used as expected.
-
-        num_calls - the number of times the call was expected (>= 0)
-        expected_messages - the send message types
-        expected_keys - the list of keys used in each message (may not all be given inspected
-          values, but all should be present)
-        expected_values - the set of values used in each message (there may be more keys, but
-          these are known key-value pairs)
-        """
-        self.assertEqual(send_comm_mock.call_count, num_calls)
-        for i in range(num_calls):
-            [message_name, message_dict] = send_comm_mock.call_args_list[i][0]
-            if expected_messages is not None:
-                self.assertEqual(message_name, expected_messages[i])
-            if expected_keys is not None:
-                for key in expected_keys[i]:
-                    self.assertIn(key, message_dict)
-            if expected_values is not None:
-                for key in expected_values[i]:
-                    self.assertEqual(message_dict.get(key), expected_values[i][key])
+    def _transform_comm_messages(self, comm_mock):
+        transformed_call_args_list = []
+        for call in comm_mock.call_args_list:
+            [msg_type, content] = call[0]
+            transformed_call_args_list.append(
+                {"msg_type": msg_type, "content": content}
+            )
+        return transformed_call_args_list
 
     def _verify_comm_error(self, comm_mock, cell_id=None, run_id=None) -> None:
         """
-        a wrapper around self._verify_comm_mock that just looks for the error message, and
-        ONLY the error message, to have been sent, so we can make these tests a little more
-        DRY.
-
-        This is really for app running cases that are expected to fail before trying to start
-        the app with EE2. Other cases should run _verify_comm_mock directly.
+        Generates standard error messages and tests it against stored comm messages.
```

Comment:
The previous version of this method and `_verify_comm_success` had a rather complex system of checking keys, values, and number of messages because unittest does not provide a nice way to ignore some keys in a dictionary. I have replaced that system with a simpler version where values that aren't easy to predict/generate are checked to be present, and then deleted; the rest of the object can be compared to the expected output.